src/Evolution/Systems/ScalarWave/BoundaryCorrections/UpwindPenalty.cpp

```diff
@@ -0,0 +1,224 @@
+    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+        char_speed_normal_times_v_minus_int,
+    const Scalar<DataVector>& char_speed_constraint_gamma2_v_psi_int,
+    const tnsr::i<DataVector, 3, Frame::Inertial>& char_speeds_int,
+
+    const Scalar<DataVector>& char_speed_v_psi_ext,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& char_speed_v_zero_ext,
+    const Scalar<DataVector>& char_speed_v_plus_ext,
+    const Scalar<DataVector>& char_speed_v_minus_ext,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+        char_speed_minus_normal_times_v_plus_ext,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+        char_speed_minus_normal_times_v_minus_ext,
+    const Scalar<DataVector>& char_speed_constraint_gamma2_v_psi_ext,
+    const tnsr::i<DataVector, 3, Frame::Inertial>& char_speeds_ext,
+    dg::Formulation /*dg_formulation*/) const noexcept {
+  const size_t num_pts = char_speeds_int[0].size();
+  Variables<tmpl::list<::Tags::TempScalar<0>, ::Tags::TempScalar<1>,
+                       ::Tags::TempScalar<2>, ::Tags::TempScalar<3>,
+                       ::Tags::TempScalar<4>, ::Tags::TempScalar<5>,
+                       ::Tags::TempScalar<6>, ::Tags::TempScalar<7>>>
+      buffer(num_pts);
+  DataVector& weighted_lambda_psi_int = get(get<::Tags::TempScalar<0>>(buffer));
+  weighted_lambda_psi_int = step_function(-char_speeds_int[0]);
+  DataVector& weighted_lambda_psi_ext = get(get<::Tags::TempScalar<1>>(buffer));
+  weighted_lambda_psi_ext = -step_function(char_speeds_ext[0]);
+
+  DataVector& weighted_lambda_zero_int =
+      get(get<::Tags::TempScalar<2>>(buffer));
+  weighted_lambda_zero_int = step_function(-char_speeds_int[0]);
+  DataVector& weighted_lambda_zero_ext =
+      get(get<::Tags::TempScalar<3>>(buffer));
+  weighted_lambda_zero_ext = -step_function(char_speeds_ext[0]);
+
+  DataVector& weighted_lambda_plus_int =
+      get(get<::Tags::TempScalar<4>>(buffer));
+  weighted_lambda_plus_int = step_function(-char_speeds_int[1]);
+  DataVector& weighted_lambda_plus_ext =
+      get(get<::Tags::TempScalar<5>>(buffer));
+  weighted_lambda_plus_ext = -step_function(char_speeds_ext[1]);
+
+  DataVector& weighted_lambda_minus_int =
+      get(get<::Tags::TempScalar<6>>(buffer));
+  weighted_lambda_minus_int = step_function(-char_speeds_int[2]);
+  DataVector& weighted_lambda_minus_ext =
+      get(get<::Tags::TempScalar<7>>(buffer));
+  weighted_lambda_minus_ext = -step_function(char_speeds_ext[2]);
+
+  // D_psi = Theta(-lambda_psi^{ext}) lambda_psi^{ext} v_psi^{ext}
+  //       - Theta(-lambda_psi^{int}) lambda_psi^{int} v_psi^{int}
+  // where the unit normals on both sides point in the same direction, out
+  // of the current element. Since lambda_psi from the neighbor is computing
+  // with the normal vector pointing into the current element in the code,
+  // we need to swap the sign of lambda_psi^{ext}. Theta is the heaviside step
+  // function with Theta(0) = 0.
```

Comment:
`with Theta(0) = 0` is inconsistent with the documentation (which gives the value at \le 0.0). I think our version of `step_function` agrees with the dox and not `Theta(0) = 0`.

Reply:
It doesn't matter in this case since we always have `Theta(Lambda) * Lambda`. Having said that, I'm not sure what you mean by "the documentation" and "the dox". The one on `UpwindPenalty` or `step_function`?

Reply:
I mean the `UpwindPenalty` dox. There it is a `\le`, where here you imply strict inequality. Perhaps just omit the part of this code comment that states the behavior at 0?